public/docs/_examples/toh-4/ts/app/app.component.1.ts

```diff
@@ -0,0 +1,63 @@
+// #docplaster
+// #docregion on-init
+import {OnInit} from 'angular2/core';
+
+// #enddocregion on-init
+import {Component} from 'angular2/core';
+import {Hero} from './hero';
+import {HeroDetailComponent} from './hero-detail.component';
+// #docregion hero-service-import
+import {HeroService} from './hero.service.1';
+// #enddocregion hero-service-import
+
+// Testable but never shown
+@Component({
+  selector: 'my-app',
+  template: `
+  <div *ngFor="#hero of heroes" (click)="onSelect(hero)">
+    {{hero.name}}
+  </div>
+  <my-hero-detail [hero]="selectedHero"></my-hero-detail>
+  `,
+  directives: [HeroDetailComponent],
+  // #docregion providers
+  providers: [HeroService]
+  // #enddocregion providers
+})
+// #docregion on-init
+export class AppComponent implements OnInit {
```

Comment:
Ah no. It was that thing where the interface is called differently than the implementation....